stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -1698,7 +1745,195 @@ fn delegate_stack_stx_signer_key() {
     .expect_tuple();
 
     let state_signer_key = stacking_state.get("signer-key").unwrap();
-    assert_eq!(state_signer_key.to_string(), signer_key_val.to_string());
+    assert_eq!(
+        state_signer_key.to_string(),
+        format!("0x{}", signer_public_key.to_hex())
+    );
+
+    stacker_nonce += 1;
+
+    let delegate_stack_extend = make_pox_4_delegate_stack_extend(
+        delegate_key,
+        stacker_nonce,
+        PrincipalData::from(key_to_stacks_addr(stacker_key)).into(),
+        pox_addr.clone(),
+        1,
+        new_signer_public_key.clone(),
+    );
+
+    let txs = vec![delegate_stack_extend];
+
+    let latest_block = peer.tenure_with_txs(&txs, &mut coinbase_nonce);
+    let new_stacking_state = get_stacking_state_pox_4(
+        &mut peer,
+        &latest_block,
+        &key_to_stacks_addr(stacker_key).to_account_principal(),
+    )
+    .unwrap()
+    .expect_tuple();
+
+    let state_signer_key_new = new_stacking_state.get("signer-key").unwrap();
+    assert_eq!(
+        state_signer_key_new.to_string(),
+        format!("0x{}", new_signer_public_key.to_hex())
+    );
+}
+
+#[test]
```

Comment:
It's worth adding a comment here that the reason this test is so spartan is because stack-increase is the same in pox-4 as it is in pox-3.  Same goes for delegate-stack-increase.